### PR TITLE
add bans and ready/unready all buttons

### PR DESCRIPTION
**Pull request by 1dNDN**

### CODE REVIEW

File: src/Main.cs [hunk 1]
```diff
@@ -59,14 +65,21 @@ private void OnGUI() {
...
 				GUILayout.EndArea();
 				return;
 			}
-
+			
+			if (GUILayout.Button("Ready all"))
+				currentPlayers.ForEach(player => SetReadyState(player, true));
+			
+			if (GUILayout.Button("Not Ready all"))
+				currentPlayers.ForEach(player => SetReadyState(player, false));
```
Comment (checkraisefold) [inline src/Main.cs]: Braces should be covered by clang-format, please run it on your code
```suggestion
			if (GUILayout.Button("Ready All")) {
				currentPlayers.ForEach(player => SetReadyState(player, true));
			}
			if (GUILayout.Button("Un-Ready All")) {
				currentPlayers.ForEach(player => SetReadyState(player, false));
			}
```

File: src/Main.cs [hunk 2]
```diff
@@ -79,16 +92,15 @@ private void OnGUI() {
 					GUILayout.Box($"Player: {player.playerName}");
 				}
 				if (player.connectionToServer == null) {
-					if (GUILayout.Button("Kick")) {
-						player.GetComponent<NetworkIdentity>().connectionToClient.Disconnect();
-					}
+					if (GUILayout.Button("Kick"))
+						KickPlayer(player);
+					if (GUILayout.Button("Ban"))
+						BanPlayer(player);
```
Comment (checkraisefold) [inline src/Main.cs]: This should be covered by clang-format, please run it on your code
```suggestion
					if (GUILayout.Button("Kick")) {
						KickPlayer(player);
					}
					if (GUILayout.Button("Ban")) {
						BanPlayer(player);
					}
```

File: src/Main.cs [hunk 3]
```diff
@@ -59,14 +65,21 @@ private void OnGUI() {
 			GUILayout.EndHorizontal();
 
 			randomizeSpawns.Value = GUILayout.Toggle(randomizeSpawns.Value, "Randomize Spawns");
-
+
```
Comment (checkraisefold) [inline src/Main.cs:68]: run clang-format

File: src/Main.cs [hunk 4]
```diff
@@ -117,5 +132,24 @@ private void Update() {
...
+		}
+		
+		private void BanPlayer(MultiplayerRoomPlayer player)
+		{
+			bannedPlayers.Add((player.netId, player.playerName));
+			KickPlayer(player);
+		}
+
+		private static void KickPlayer(MultiplayerRoomPlayer player) =>
+			player.GetComponent<NetworkIdentity>().connectionToClient.Disconnect();
```
Comment (checkraisefold) [inline src/Main.cs:153]: Run clang-format for braces please this should not be an arrow function

File: src/Main.cs [hunk 5]
```diff
@@ -117,5 +132,24 @@ private void Update() {
...
+		
+		private bool IsPlayerBanned(MultiplayerRoomPlayer player) =>
+			bannedPlayers.Any(ban => ban.Item1 == player.netId) || bannedPlayers.Any(ban => ban.Item2 == player.playerName);
```
Comment (checkraisefold) [inline src/Main.cs]: I would bet money that netId doesn't stay the same, and name also isn't reliable

Mirror's `Transport::ServerGetClientAddress` converts a connection ID to a SteamID with the FizzyFacepunch transport the game uses, so use that for a good identifier. Use `MultiplayerRoomManager::GetTransport` to get that transport object, earlier in the Main file I get it using the singleton

Also run clang-format

**Review by checkraisefold — CHANGES_REQUESTED** [review]

**Comment by checkraisefold** [comment]
Also pull latest changes from main so the workflow can run without blowing up

**Comment by 1dNDN** [comment]
I fixed the style manually, I don't have a clang-format, and my IDE has a very different style set up

File: src/Main.cs (hunk 5 above)
Comment (1dNDN) [reply src/Main.cs]: I looked in the source code, netid is a really bad idea, but the nickname rarely changes

**Comment by checkraisefold** [comment]
fixing this manually

**Pull request merged**